src/coreclr/hosts/inc/coreclrhost.h

```diff
@@ -19,9 +19,15 @@
 // For each hosting API, we define a function prototype and a function pointer
 // The prototype is useful for implicit linking against the dynamic coreclr
 // library and the pointer for explicit dynamic loading (dlopen, LoadLibrary)
+#ifdef __cplusplus
 #define CORECLR_HOSTING_API(function, ...) \
     extern "C" int CORECLR_CALLING_CONVENTION function(__VA_ARGS__); \
```

Comment:
nit: there is `EXTERN_C` macro to avoid the condition.

Reply:
This header is meant for external consumption and it cannot have other dependencies.

Reply:
Ah ok, maybe this header could use the same preamble as the other shared headers:

```h
#ifdef __cplusplus
extern "C"
{
#endif // __cplusplus

... code ...

#ifdef __cplusplus
}
#endif // __cplusplus
```

Reply:
I have no personal preference for either style. They are nearly the same in terms of changed lines.